dlt/destinations/decorators.py

```diff
@@ -0,0 +1,26 @@
+from typing import Any, Type
+from dlt.destinations.impl.destination.factory import destination as _destination
+from dlt.destinations.impl.destination.configuration import SinkClientConfiguration, TSinkCallable
+from dlt.common.destination import TDestinationReferenceArg
+from dlt.common.destination import TLoaderFileFormat
+
+
+def destination(
+    loader_file_format: TLoaderFileFormat = None,
+    batch_size: int = 10,
+    name: str = None,
+    naming_convention: str = "direct",
+    spec: Type[SinkClientConfiguration] = SinkClientConfiguration,
+) -> Any:
+    def decorator(destination_callable: TSinkCallable) -> TDestinationReferenceArg:
```

Comment:
Decorator returns `Destination` but the user should see the signature of the decorated function as return type, We''ll add `__call__` to Destination to simulate it being a function. this is same trick we do with DltResource

I can do the above if you want

Reply:
Ah, now I think I get what you want here. Yes why not, maybe you can add that one small change

Reply:
Actually I think to me it is not quite clear what i should do here, i changed the decorator typings a bit and made it so a wrapped function is returned with the args in place (i think at least) but maybe you need to make some changes here.